x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/helpers.ts

```diff
@@ -71,24 +71,47 @@ export const getKBVectorSearchQuery = ({
       ]
     : [];
 
-  const userFilter = [
-    {
-      nested: {
-        path: 'users',
-        query: {
-          bool: {
-            must: [
-              {
-                match: user.profile_uid
-                  ? { 'users.id': user.profile_uid }
-                  : { 'users.name': user.username },
-              },
-            ],
+  const userFilter = {
```

Comment:
Query bug found when writing integration tests. This updates to ensure `global` entires (where user[] is empty/missing) are returned along with the current user's entries.